client/src/components/ProductDetail/ProductDetailInfo/AmountPanel.tsx

```diff
@@ -0,0 +1,133 @@
+import styled from '@emotion/styled';
+import React from 'react';
+
+const AmountPanelWrapper = styled.div`
+  width: 100%;
+  padding: 20px;
+  display: flex;
+  justify-content: space-between;
+  background-color: #f5f5f5;
+  box-sizing: border-box;
+  margin-top: 1.5rem;
+`;
+
+const AmountPanelLeft = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 13px;
+  font-weight: 500;
+`;
+
+const AmountPanelRight = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: right;
+`;
+
+const AmountCounter = styled.div`
+  width: 99px;
+  display: flex;
+  justify-content: center;
+  height: 1.5em;
+  padding: 0 17px;
+`;
+
+const AmountCounterInput = styled.input`
+  width: 2rem;
+  text-align: center;
+  outline: none;
+  border: 1px solid #ccc;
+`;
+
+const IDCreaseButtonWrapper = styled.div`
+  padding-left: 5px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-top: 4px;
+  flex-direction: column;
+`;
+
+const IDcreaseButton = styled.span`
+  font-size: 16px;
+  font-weight: bold;
+  line-height: 100%;
+  cursor: pointer;
+  width: 1.5rem;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const PriceDisplay = styled.div`
+  display: flex;
+  justify-content: right;
+  align-items: center;
+`;
+const DeleteButton = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-left: 1rem;
+  width: 1rem;
+  height: 1rem;
+`;
+
+export interface IAmountControl {
+  amount: number;
+  setAmount: Function;
+}
+
+export interface IAmountPanelProps {
+  amountControl: IAmountControl;
+  optionName: string | undefined;
+  price: number;
+}
+
+const AmountPanel = (props: IAmountPanelProps) => {
+  const { amountControl, price, optionName } = props;
+  const { amount, setAmount } = amountControl;
+
+  const countBtnClickHandler = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    const { counthop } = event.currentTarget.dataset;
+    if (counthop === undefined) {
+      return;
+    }
+
+    const changeToAmount = amount + +counthop <= 1 ? 1 : amount + +counthop;
+    setAmount(changeToAmount);
+  };
```

Comment:
```suggestion
  const countBtnClickHandler = (add: number) => () => {
    const changedAmount = amount + add > 0 ? amount + add : 1;
    setAmount(changedAmount);
  };
```

이런 방식으로도 표현해볼 수 있을 것 같아요. 😄
만약 사용하신다면 네이밍은 바꾸는 걸로 😅

Reply:
오 좋네요... 반영하였습니다 1